hugr/src/hugr.rs

```diff
@@ -196,6 +196,7 @@ impl Hugr {
         extension_registry: &ExtensionRegistry,
     ) -> Result<(), ValidationError> {
         resolve_extension_ops(self, extension_registry)?;
+        self.validate_no_extensions(extension_registry)?;
```

Comment:
What I'd expect to see here is - so far so good exactly this - but after `validate_no_extensions` can we condition on `#[cfg(feature="extension_inference")]` and then do
* If extension inference is on, infer extensions, and then validate *just* extensions (there's no need to repeat the rest of validation, is there?)
* If extension inference is off....nothing more to do since we've validated everything else?